radix-engine/src/vm/wasm/wasmi.rs

```diff
@@ -109,30 +109,51 @@ fn consume_buffer(
     }
 }
 
-fn actor_call_module_method(
+fn call_method(
     mut caller: Caller<'_, HostState>,
-    module_id: u32,
+    receiver_ptr: u32,
+    receiver_len: u32,
     ident_ptr: u32,
     ident_len: u32,
     args_ptr: u32,
     args_len: u32,
 ) -> Result<u64, InvokeError<WasmRuntimeError>> {
     let (memory, runtime) = grab_runtime!(caller);
 
+    let receiver = read_memory(caller.as_context_mut(), memory, receiver_ptr, receiver_len)?;
     let ident = read_memory(caller.as_context_mut(), memory, ident_ptr, ident_len)?;
     let args = read_memory(caller.as_context_mut(), memory, args_ptr, args_len)?;
 
     runtime
-        .actor_call_module_method(module_id, ident, args)
+        .object_call(receiver, ident, args)
         .map(|buffer| buffer.0)
 }
 
-fn call_method(
+fn call_direct_method(
+    mut caller: Caller<'_, HostState>,
+    receiver_ptr: u32,
+    receiver_len: u32,
+    ident_ptr: u32,
+    ident_len: u32,
+    args_ptr: u32,
+    args_len: u32,
+) -> Result<u64, InvokeError<WasmRuntimeError>> {
+    let (memory, runtime) = grab_runtime!(caller);
+
+    let receiver = read_memory(caller.as_context_mut(), memory, receiver_ptr, receiver_len)?;
+    let ident = read_memory(caller.as_context_mut(), memory, ident_ptr, ident_len)?;
+    let args = read_memory(caller.as_context_mut(), memory, args_ptr, args_len)?;
+
+    runtime
+        .object_call_direct(receiver, ident, args)
+        .map(|buffer| buffer.0)
+}
+
+fn call_module_method(
     mut caller: Caller<'_, HostState>,
     receiver_ptr: u32,
     receiver_len: u32,
     direct_access: u32,
```

Comment:
direct_access?

Reply:
good catch! surprised we don't have a test catching this. Will add one if missing